assets/model_monitoring/components/src/feature_importance_metrics/compute_feature_importance.py

```diff
@@ -220,9 +229,71 @@ def write_to_mltable(explanations, dataset, file_path, categorical_features):
                                          constants.METRIC_NAME_COLUMN,
                                          constants.FEATURE_CATEGORY_COLUMN,
                                          constants.THRESHOLD_VALUE], index=[0])
-    print(explanation_data)
     metrics_data = pd.concat(explanation_data)
-    spark_data = convert_pandas_to_spark(metrics_data)
+    return metrics_data
+
+
+def create_local_feature_importance_df(explanations, dataset, task_type):
+    """Write local feature importance values to mltable.
+
+    :param explanations: list of local feature importances in the order of the baseline columns
+    :type explanations: list[float]
+    :param dataset: dataset to derive feature names
+    :type dataset: pandas.Dataframe
+    :param task_type: the task type (regression or classification) of the resulting model
+    :type task_type: string
+    :return: local feature importance sample table
+    :rtype: pandas.Dataframe
+    """
+    # Create column names
+    columns = [constants.ROW_INDEX]
+    if task_type == constants.CLASSIFICATION:
+        columns.append(constants.CLASS)
+    feature_names = list(dataset.columns)
+    columns += feature_names
+
+    sample_data = pd.DataFrame(columns=columns)
+
+    # Populate importances
+    for index in range(len(explanations)):
+        new_row = {}
+        # TODO: Does this need to be set to some correlation ID beyond the index of the row?
+        new_row[constants.ROW_INDEX] = index
+
+        if task_type == constants.CLASSIFICATION:
+            for potential_class in range(len(explanations[index])):
+                # TODO: Class is currently the index of the class in the local explanations matrix,
```

Comment:
is this TODO still valid ?

Reply:
Yes - would require changes to RAI Insights explainer_manager to access class names. I added a long comment about it in the PR description (not sure if that is the right place for it so let me know if I should move it).